src/cursor/cur_hs.c

```diff
@@ -253,17 +260,23 @@ __curhs_reset(WT_CURSOR *cursor)
     ret = file_cursor->reset(file_cursor);
     WT_TIME_WINDOW_INIT(&hs_cursor->time_window);
     hs_cursor->btree_id = 0;
-    hs_cursor->datastore_key.data = NULL;
-    hs_cursor->datastore_key.size = 0;
+    hs_cursor->datastore_key->data = NULL;
+    hs_cursor->datastore_key->size = 0;
     hs_cursor->flags = 0;
+    cursor->key.data = cursor->key.mem;
+    cursor->key.size = cursor->key.memsize;
+    cursor->value.data = cursor->value.mem;
+    cursor->value.size = cursor->value.memsize;
+    F_CLR(cursor, WT_CURSTD_KEY_SET);
+    F_CLR(cursor, WT_CURSTD_VALUE_SET);
 
 err:
     API_END_RET(session, ret);
 }
 
 /*
  * __curhs_set_key --
- *     WT_CURSOR->set_key method for the hs cursor type.
+ *     WT_CURSOR->set_key method for the history store cursor type.
  */
 static void
 __curhs_set_key(WT_CURSOR *cursor, ...)
```

Comment:
Where's the code where you're using the explicitly passed-in argument count? I was interested in reading that part but couldn't find it.

Reply:
I have tagged you to the correct code line.